### PR TITLE
Compat.isapprox method with nans keyword argument

### PR DESCRIPTION
**Pull request by andreasnoack**

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -947,6 +947,16 @@ if VERSION < v"0.4.0-dev+6578"
     ≉(x,y) = !(x ≈ y)
     export ≈, ≉
 end
+if VERSION < v"0.6.0-dev.2093" # Compat.isapprox to allow for NaNs
+    if VERSION >= v"0.4.0-dev+6578"
```
Comment (martinholters) [inline src/Compat.jl]: We require `julia 0.4`, so this should always be true?

Comment (andreasnoack) [inline src/Compat.jl]: Thanks. Then I might as well just remove the previous definition as part of this PR.

File: src/Compat.jl [hunk 2]
```diff
@@ -1774,4 +1761,13 @@ if VERSION >= v"0.5.0-dev+5509" && VERSION < v"0.6.0-dev.1614"
     include_string(".|(xs...) = broadcast(|, xs...)")
 end
 
+if VERSION < v"0.6.0-dev.2093" # Compat.isapprox to allow for NaNs
+    using Base.rtoldefault
+    function isapprox(x::Number, y::Number; rtol::Real=rtoldefault(x,y), atol::Real=0, nans::Bool=false)
```
Comment (tkelman) [inline src/Compat.jl]: don't want to overwrite the default when nans is not specified, right?

Comment (andreasnoack) [inline src/Compat.jl]: Yes. That was the plan.

Comment (tkelman) [inline src/Compat.jl]: is it possible to define a kwarg method without a default value?

Comment (ararslan) [inline src/Compat.jl]: Nope, that's a syntax error saying that the kwarg needs a default value.

**Comment by andreasnoack** [comment]
Can we merge this? The tests are failing because of the `wakedir` breakage.

File: src/Compat.jl (hunk 2 above)
Comment (tkelman) [inline src/Compat.jl]: Let me phrase the original question differently then. Doesn't the default-value method here overwrite the base method without the nan kwarg?

Comment (andreasnoack) [inline src/Compat.jl]: I'm not importing the Base method so I wouldn't think so.

**Pull request merged**